File: q10-q19/q19/q19.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dum_head = new ListNode;
        dum_head->next = head;
        ListNode* fast = dum_head;
        ListNode* slow = dum_head;
        for (int i = 0; i < n+1; i++){
            fast = fast->next;
        }
        while (fast != NULL){
            fast = fast->next;
            slow = slow->next;
        }
        slow->next = slow->next->next;
        return dum_head->next;
    }
};

File: q1-q99/q10-q19/q19/q19.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        dum_head = ListNode();
        dum_head.next = head;
        fast = dum_head;
        slow = dum_head;
        for i in range(n):
            fast = fast.next;
        while fast.next != None:
            fast = fast.next;
            slow = slow.next;
        slow.next = slow.next.next;
        return dum_head.next;
